Sources/Sapi.php

```diff
@@ -0,0 +1,318 @@
+		self::SERVER_IIS => 'Microsoft-IIS',
+		self::SERVER_APACHE => 'Apache',
+		self::SERVER_LITESPEED => 'LiteSpeed',
+		self::SERVER_LIGHTTPD => 'lighttpd',
+		self::SERVER_NGINX => 'nginx'
+	];
+
+	/*****************
+	 * Class constants
+	 *****************/
+
+	public const SERVER_IIS = 'iis';
+	public const SERVER_APACHE = 'apache';
+	public const SERVER_LITESPEED = 'litespeed';
+	public const SERVER_LIGHTTPD = 'lighttpd';
+	public const SERVER_NGINX = 'nginx';
+
+	public const OS_WINDOWS = 'Windows';
+	public const OS_MAC = 'Darwin';
+	public const OS_LINUX = 'linux';
+
+	/***********************
+	 * Public static methods
+	 ***********************/
+
+	/**
+	 * Checks if the server is running a specific server software.
+	 * 
+	 * @param string|string[] $server The server software as defined in our constants.
+	 * @return bool True if we are running the requested software, false otherwise.
+	 */
+	public static function isSoftware(string|array $server): bool
+	{
+		$servers = (array) $server;
+
+		foreach ($servers as $server) {
+			if (
+				isset($_SERVER['SERVER_SOFTWARE'])
+				&& strpos($_SERVER['SERVER_SOFTWARE'], self::$server_software[$server] ?? $server) !== false	
+			) {
+				return true;
+			}	
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if the server is running a specific operating system.
+	 * 
+	 * @param string|string[] $os The os as defined in our constants
+	 * @return bool True if we are running the requested os, false otherwise.
+	 */
+	public static function isOS(string|array $os): bool
+	{
+		$oses = (array) $os;
+
+		/** 
+		 * Technically we could simplify this down using PHP_OS_FAMILY,
+		 * but to ensure backwards compatibility, we won't yet.
+		 */
+		foreach ($oses as $os) {
+			switch ($os) {
+				case self::OS_WINDOWS:
+					$is_os = PHP_OS_FAMILY === 'Windows' || DIRECTORY_SEPARATOR === '\\';
+					break;
+				case self::OS_MAC:
+					$is_os = PHP_OS_FAMILY === 'Darwin';
+					break;
+				// This may result in false positives because 'linux' is very broad.
+				case self::OS_LINUX:
+					$is_os = PHP_OS_FAMILY === 'Linux';
+					break;
+			}
+
+			if ($is_os) {
+				return true;
+			}
+		}
+
+		// Last ditch effort.
+		return PHP_OS_FAMILY === $os;
+	}
+
+	/**
+	 * Checks If we are runnig a CGI instance.
+	 * 
+	 * @return bool True if we are running under CGI, false otherwise.
+	 */
+	public static function isCGI(): bool
+	{
+		return isset($_SERVER['SERVER_SOFTWARE']) && strpos(php_sapi_name(), 'cgi') !== false;
+	}
+
+	/**
+	 * Checks If we are runnig a CLI (shell/cron) instance.
+	 * 
+	 * @return bool True if we are running under CLI, false otherwise.
+	 */
+	public static function isCLI(): bool
+	{
+		return empty($_SERVER['REQUEST_METHOD']);
+	}
+
+	/**
+	 * Checks if the server is able to support case folding.
+	 * 
+	 * @see https://www.w3.org/TR/charmod-norm/#definitionCaseFolding
+	 * @return bool True if it does, false otherwise
+	 */
+	public static function isCaseFolding(): bool
```

Comment:
```suggestion
	public static function supportsIsoCaseFolding(): bool
```

This name would be clearer, in my opinion,